src/components/molecules/NavBar/NavBar.tsx

```diff
@@ -197,10 +201,16 @@ export const NavBar: React.FC<NavBarPropsType> = ({
     []
   );
 
-  if (!venueId || !currentVenue) return null;
+  if (!ALLOW_NO_VENUE && !(venueId && currentVenue)) {
+    console.warn(
+      NavBar.name,
+      `aborted display because of missing id (${venueId}) or venue (${currentVenue})`
+    );
+    return null;
+  }
 
   // TODO: ideally this would find the top most parent of parents and use those details
-  const navbarTitle = parentVenue?.name ?? currentVenue.name;
+  const navbarTitle = parentVenue?.name ?? currentVenue?.name;
```

Comment:
`currentVenue?.name` We need to check and render the code from here on downwards only if `currentVenue` exists otherwise the radios, labels, searching and a lot more won't function like it should. Either our types are off or the guard at the top is not valid.

Reply:
(So the search will return 0 items 🤷)

Last time I checked, there were like 2 components that needed `venueId` but worked OK with it being empty string. 
NavBar isn't visibly broken, and if it happens with some concurrent PR, there's a handy flag that will return the previous behavior until fixed.

Reply:
If it was guarded, it was guarded for a reason. Sometimes it looks stupid and simple and you are thinking I can remove this it's not a big real but then we have huge regression issues like the flashing of the 404 screen. We should either add the guard back or at least add another fallback condition with `??`

Reply:
> If it was guarded, it was guarded for a reason.

That's why I checked them before adding a flag to make it easily revertable. I did my due diligence. Components didn't break, NavBar allowed users to log out or go back.

Reply:
The difference is this:

No `NavBar`

vs

`NavBar` that doesn't' display title because there isn't  a venue, but does display functional components, even if the Search might not return anything when searching on a non-existent venue.

Reply:
> but then we have huge regression issues like the flashing of the 404 screen

☝️ Not huge. Not regression. That was legacy code left to do it's original job on purpose. Purpose being: smaller and faster PRs, not doing OOS changes.

Reply:
@denisdimitrov did you check where `navbarTitle` is actually used?

Reply:
@sunny-viktoryia have you considered the possibility both <NotFound /> and <NavBar /> are being shown?

Reply:
Since it is not about development taking sides, merging something, product input overriding, looped product team earlier to provide desirable resolution https://doesitsparkle.slack.com/archives/C01700P4QCT/p1631548197056100?thread_ts=1631516472.052000&cid=C01700P4QCT

Reply:
Now using `hasVenue` check for individual components as per the slack thread